### PR TITLE
zwave value check for signed/unsigned parameters

### PR DESCRIPTION
**Pull request by MatthewWaanders**

### CODE REVIEW

File: lib/App/index.js [hunk 1]
```diff
@@ -189,6 +189,25 @@ class App {
 						}
 					}
 				}
+
+                if (driver.zwave) {
+                    for (let j = 0; j < driver.settings.length; j++) {
+                        let setting = driver.settings[j];
+                        let signed;
+                        let size = setting.zwave.size;
```
Comment (WeeJeWel) [inline lib/App/index.js]: Can throw if setting.zwave is not set

File: lib/App/index.js [hunk 2]
```diff
@@ -189,6 +189,25 @@ class App {
...
+                        let max = setting.attr.max;
```
Comment (WeeJeWel) [inline lib/App/index.js]: ( setting.attr && setting.attr.max ) || setting.max

Comment (WeeJeWel) [reply lib/App/index.js]: What if max is not set?

File: lib/App/index.js (hunk 1 above)
Comment (RobinBol) [inline lib/App/index.js]: Thus might throw ‘size is undefined’ if a zwave driver implements a non zwave setting.

File: lib/App/index.js (hunk 2 above)
Comment (RobinBol) [inline lib/App/index.js]: Same here, not all settings are of type number and have an attr property.

File: lib/App/index.js [hunk 3]
```diff
@@ -189,6 +189,25 @@ class App {
...
+
+                        if (typeof setting.zwave.signed !== 'undefined') {
+                            signed = setting.zwave.signed;
```
Comment (RobinBol) [inline lib/App/index.js]: Same, zwave property might not exist.

File: lib/App/index.js [hunk 4]
```diff
@@ -189,6 +189,27 @@ class App {
...
 				}
+
+                if (driver.zwave) {
+                    for (let j = 0; j < driver.settings.length; j++) {
+                        let setting = driver.settings[j];
+                        if (setting.zwave && settings.attr && settings.attr.max) {
+                            let signed;
+                            let size = setting.zwave.size;
+                            let max = setting.attr.max;
+
```
Comment (WeeJeWel) [inline lib/App/index.js]: Check if boolean

File: lib/App/index.js [hunk 5]
```diff
@@ -189,6 +189,27 @@ class App {
...
+                            if (typeof setting.zwave.signed !== 'undefined') {
+                                signed = setting.zwave.signed;
+                            } else {
+                                signed = true;
+                            }
+
+                            if (size === 1 && max > 127 && max < 255 && signed ||
+                                size === 2 && max > 32767 && max < 65535 && signed ||
+                                size === 4 && max > 2147483647 && max < 4294967295 && signed) throw new Error(`Value cannot be signed: ${driver}, ${setting}`);
```
Comment (WeeJeWel) [inline lib/App/index.js]: throw on a new line with tab

File: lib/App/index.js (hunk 4 above)
Comment (MatthewWaanders) [reply lib/App/index.js]: fixed

File: lib/App/index.js [hunk 6]
```diff
@@ -189,6 +189,28 @@ class App {
 						}
 					}
 				}
+
+                if (driver.zwave) {
+                    for (let j = 0; j < driver.settings.length; j++) {
```
[the remark this answers is not in the document]
Comment (MatthewWaanders) [reply lib/App/index.js]: Yes this has to be an array, checked by the json schema

**Pull request merged**